src/framework/execution/types/context.ts

```diff
@@ -2,23 +2,28 @@ import { IntegrationInstance, IntegrationInstanceConfig } from './instance';
 import { JobState } from './jobState';
 import { IntegrationLogger } from './logger';
 
+export interface ExecutionContext {
+  logger: IntegrationLogger;
+}
+
 /**
  * @param TConfig the integration specific type of the `instance.config`
  * property
  */
-export interface IntegrationExecutionContext<
+export type IntegrationExecutionContext<
```

Comment:
I'm still a bit fuzzy on the differences between `type` and `interface`, but I think this change will prevent the use of `T extends IntegrationExecutionContext` right? Any reason to think that would never be something folks should do?

Reply:
You can extend `types` (see [this playground](https://www.typescriptlang.org/play/?ssl=1&ssc=1&pln=10&pc=2#code/C4TwDgpgBAKlC8UDeUAeAuKA7ArgWwCMIAnKAXwCgKBLLYEgMwEMBjaAISglXqwBMAzrGRQKUcVBCZchEhUoUWAeywDgUApk6IkYiRigBGADR7xUo-KA)). Older versions of typescript did not support it though (see [docs](https://www.typescriptlang.org/docs/handbook/advanced-types.html#interfaces-vs-type-aliases)).

It seems like the only real difference between an `interface` and `type` is that an `interface` allows for declaration merging (see [example](https://www.typescriptlang.org/play/?ssl=15&ssc=2&pln=1&pc=1#code/C4TwDgpgBAKlC8UDeUAeAuKA7ArgWwCMIAnKAXwCgKBLLYEgMwEMBjaAISglXqwBMAzrGRQKUcVBCZchEhUo06jVhxFiJAL2n4ixeVRYB7LAOBQCmToiTrxGKAFYANLcmZnrrY-lA))

Reply:
I see. I wonder if that declaration merging feature is a good thing or not. Reminds me of re-opening classes across files, or monkey-patching interfaces one does not own. I appreciate the power of being able to do what is necessary, but it comes with so much responsibility 😄